meson.build

```diff
@@ -0,0 +1,350 @@
+## SPDX-License-Identifier: 0BSD
+## Author: Vincent Torri
+
+project(
+    'xz',
+    'c',
+    version: '5.6.99',
+    license: '0BSD',
+    license_files: [
+        'COPYING',
+        'COPYING.0BSD',
+        'COPYING.GPLv2',
+        'COPYING.GPLv3',
+        'COPYING.LGPLv2.1'
+    ],
+    default_options: [
+        'b_ndebug=if-release',
+        'c_std=c11',
+        'warning_level=3',
+    ],
+    meson_version: '>= 1.1.0',
+)
+
+v_array = meson.project_version().split('.')
+v_maj = v_array[0]
+
+# config.h
+
+config_dir = [include_directories('.')]
+
+config_h = configuration_data()
+config_h.set_quoted('PACKAGE', 'xz')
+config_h.set_quoted('PACKAGE_BUGREPORT', 'xz@tukaani.org')
+config_h.set_quoted('PACKAGE_NAME', meson.project_name())
+config_h.set_quoted('PACKAGE_URL', 'https://tukaani.org/xz/')
+
+# host
+
+config_h.set10('WORDS_BIGENDIAN', host_machine.endian() == 'big',
+    description: 'Define to 1 if the processor stores words with the most significant byte first.'
+)
+
+host_os = host_machine.system()
+
+cygwin = 'cygwin'
+windows = 'windows'
+linux = 'linux'
+sunos = 'sunos'
+asm_os = [ 'linux', 'dragonfly', 'freebsd', 'netbsd', 'openbsd', 'windows', 'cygwin' ]
+sys_cygwin = cygwin.contains(host_os)
+sys_windows = windows.contains(host_os)
+
+COND_W32 = sys_cygwin or sys_windows
+
+# binaries
+cc = meson.get_compiler('c')
+
+# try mimic AC_USE_SYSTEM_EXTENSIONS
+if host_machine.system() == 'linux'
+    add_project_arguments('-D_GNU_SOURCE', language: 'c')
```

Comment:
Autoconf defines `_GNU_SOURCE` and the few other `AC_USE_SYSTEM_EXTENSIONS` macros unconditionally and it seems to be fine in practice. So I suggest simplifying this. Then GNU on something else than Linux (like HURD) will work without any extra effort too.

Reply:
done in fb6b458e9ed311ed1728de3b4c4617fd8d5861d3